catalog/ddms/catalog-ddms-transformer/src/main/java/org/codice/alliance/ddms/v2/Ddms20InputTransformer.java

```diff
@@ -0,0 +1,762 @@
+    metacard.setAttribute(getPublisherNames(ddms));
+    metacard.setAttribute(getPublisherEmails(ddms));
+    metacard.setAttribute(getPublisherPhones(ddms));
+    metacard.setAttribute(getContributorNames(ddms));
+    metacard.setAttribute(getContributorEmails(ddms));
+    metacard.setAttribute(getContributorPhones(ddms));
+    metacard.setAttribute(getPointOfContactNames(ddms));
+    metacard.setAttribute(getPointOfContactEmails(ddms));
+    metacard.setAttribute(getPointOfContactPhones(ddms));
+
+    // Location attributes
+    metacard.setAttribute(getCountryCodes(ddms));
+    metacard.setAttribute(getCrsNames(ddms));
+    // TODO: location.altitude-meters is a content extension
+
+    // Media attributes
+    metacard.setAttribute(getMediaFormat(ddms));
+    metacard.setAttribute(getMediaType(ddms));
+
+    // Topic attributes
+    metacard.setAttribute(getTopicCategories(ddms));
+    metacard.setAttribute(getTopicKeywords(ddms));
+    metacard.setAttribute(getTopicVocabulary(ddms));
+
+    // @TODO: isr.frequency-hertz is a content extension
+
+    // Security attributes
+    SecurityAttributes securityAttributes = ddms.getSecurity().getSecurityAttributes();
+    metacard.setAttribute(getClassification(securityAttributes));
+    metacard.setAttribute(getReleasability(securityAttributes));
+    metacard.setAttribute(getOwnerProducers(securityAttributes));
+    metacard.setAttribute(getDisseminationControls(securityAttributes));
+    metacard.setAttribute(getSciControls(securityAttributes));
+    metacard.setAttribute(getNonIcMarkings(securityAttributes));
+
+    // Restriction attributes
+    metacard.setAttribute(getCopyright(ddms));
+    metacard.setAttribute(getIntellectualProperty(ddms));
+    metacard.setAttribute(getPrivacyAct(ddms));
+
+    // Ddms attributes
+    metacard.setAttribute(getInfoCutOff(ddms));
+    metacard.setAttribute(getPosted(ddms));
+    metacard.setAttribute(getGeographicNames(ddms));
+    metacard.setAttribute(getGeographicRegions(ddms));
+    metacard.setAttribute(getBeNumbers(ddms));
+    metacard.setAttribute(getSubtitles(ddms));
+    // TODO: ext.ddms.icid-payload
+    // TODO: ext.ddms.security portion markings
+    // Portion-Markings: Title, Subtitle, description, contacts, related resources, security
+    // TODO: ext.ddms.related-resource not used?
+    // Todo: Not sure what ext.ddms.topic-type is for
+
+    return metacard;
+  }
+
+  private DdmsResource parseDdms(InputStream input) throws CatalogTransformerException {
+    try {
+      XMLStreamReader xmlStreamReader = XMLInputFactory.newFactory().createXMLStreamReader(input);
+      return new Ddms20XmlReader(xmlStreamReader).read();
+    } catch (XMLStreamException e) {
+      throw new CatalogTransformerException("Failed to parse DDMS 2.0", e);
+    }
+  }
+
+  private Attribute getTitle(DdmsResource ddms) {
+    String title = ddms.getTitles().stream().findFirst().map(Title::getValue).orElse(null);
```

Comment:
can ddms.getTitles() be null?

Reply:
Not positive, but I think title is a required ddms element. I'll do a once through to make sure we're not missing any null checks that are needed, though

Reply:
Turns out most of the fields on the DdmsResource interface are non-nullable. See https://github.com/codice/openddms/blob/master/src/main/kotlin/org/codice/ddms/DdmsResource.kt.

I confirmed the fields that are nullable have the appropriate null checks